compass/landice/tests/ensemble_generator/plot_ensemble.py

```diff
@@ -202,61 +310,149 @@
+            qoi_info['grd area change']['values'][idx] = (grdArea[ii] -
+                                                          grdArea[0])
+            qoi_info['grd area change rate']['values'][idx] = \
+                (grdArea[ii] - grdArea[jj]) / rate_dt
+            qoi_info['total area change']['values'][idx] = (iceArea[ii] -
+                                                            iceArea[0])
+            qoi_info['total area change rate']['values'][idx] = \
+                (iceArea[ii] - iceArea[jj]) / rate_dt
+            qoi_info['GL flux']['values'][idx] = groundingLineFlux[ii]
+            qoi_info['melt flux']['values'][idx] = BMB[ii]
 
-        # plot time series
-        axSLRts.plot(years, SLR, linewidth=lw, color=col, alpha=alph)
-        axTAts.plot(years, totalArea - totalArea[0], linewidth=lw,
-                    color=col, alpha=alph)
-        axGAts.plot(years, grdArea - grdArea[0], linewidth=lw,
-                    color=col, alpha=alph)
-        axFAts.plot(years, fltArea, linewidth=lw, color=col, alpha=alph)
-        # ignore first entry which is 0
-        axGLFts.plot(years[1:], groundingLineFlux[1:], linewidth=lw,
-                     color=col, alpha=alph)
-        # ignore first entry which is 0
-        axBMBts.plot(years[1:], BMB[1:], linewidth=lw,
-                     color=col, alpha=alph)
-
-        # Only process runs that have reached target year
-        indices = np.nonzero(years >= targetYear)[0]
+            # filter run
+            valid_run = filter_run()
+            if valid_run:
+                n_filtered += 1
+        else:
+            valid_run = False
+
+        # calculate qoi's requiring spatial output
+        DS = xr.open_mfdataset(run + '/output/' + 'output_*.nc',
+                               combine='nested', concat_dim='Time',
+                               decode_timedelta=False,
+                               chunks={"Time": 10})
+        yearsOutput = DS['daysSinceStart'].values[:] / 365.0
+        indices = np.nonzero(yearsOutput >= target_year)[0]
         if len(indices) > 0:
             ii = indices[0]
-            print(f'{run} using year {years[ii]}')
-            qoi_info['SLR']['values'][idx] = SLR[ii]
-
-            grdArea = f.variables['groundedIceArea'][:] / 1000.0**2  # in km^2
-            qoi_info['grd area']['values'][idx] = grdArea[ii] - grdArea[0]
 
-            iceArea = f.variables['totalIceArea'][:] / 1000.0**2  # in km^2
-            qoi_info['total area']['values'][idx] = iceArea[ii] - iceArea[0]
-
-            qoi_info['GL flux']['values'][idx] = groundingLineFlux[ii]
+            surfaceSpeed = DS['surfaceSpeed'].values[ii, :]
+            thickness = DS['thickness'].values[ii, :]
+            cellMask = DS['cellMask'].values[ii, :]
+            areaCell = DS['areaCell'].values[0, :]
+
+            # only evaluate where modeled ice exists and observed speed
+            # uncertainy is a reasonable value (<~1e6 m/yr)
```

Comment:
is 1e6 m/yr a reasonable value?

Reply:
That was screening out some no data values.  Improving on this robustly would best include handling no data values in netcdf files and in the code explicitly.